client/src/pages/trainer/courses/[id]/index.tsx

```diff
@@ -35,6 +39,15 @@ const CourseContent: React.FC = () => {
     }
   }, [course]);
 
+  const newError = error as typeof error & {
+    status: number | string;
+  };
+
+  if (isError && (newError.status === 404 || newError.status === 'PARSING_ERROR')) {
+    void replace('/404');
+    return;
+  }
+
```

Comment:
I was expecting this implementation to be done on server side  using `getServerSideProps`, that is also what the most recommends but since you are using this approach, please try to remove the flicker on FE

https://github.com/framgia/sph-lms/assets/110364690/8574fc9c-cf51-4eba-8b88-da22edc0b631